src/Injector.php

```diff
@@ -26,20 +26,13 @@
  */
 class Injector implements InjectorInterface
 {
-    /** @var DefinitionInterface */
-    protected $definition;
-
-    /** @var ContainerInterface */
-    protected $container;
-
-    /** @var DependencyResolverInterface */
-    protected $resolver;
-
-    /** @var ConfigInterface */
-    protected $config;
+    protected DefinitionInterface $definition;
```

Comment:
This type addition is a BC break for inheritance scenarios 🤔

Reply:
Back to comment is okay

Reply:
On a side note, is there a way for me to detect BC breaks like those up front?

Reply:
I wrote https://github.com/Roave/BackwardCompatibilityCheck, which is somewhat able to detect some of these :-) 

I'd really like to integrate it with our CI: https://github.com/laminas/laminas-ci-matrix-action/issues/60